bionic/task_state.py

```diff
@@ -168,6 +179,26 @@ def _compute(self, task_key_logger):
 
         dep_values = [dep_result.value for dep_result in dep_results]
 
+        # If we have any missing outputs, exit early with a missing result.
+        if self.has_missing_output():
```

Comment:
I think this is fine here for now but we should consider moving this logic for entries that exit early like this one and simple lookups somewhere else.

Reply:
Simple lookups don't exit early, right? They get handled by `task.compute` like everything else.

That said, yes, as we add more "generic" rules, like with tuples, we should think about how to handle them in a uniform way.